editor/src/core/commenting/comment-types.tsx

```diff
@@ -0,0 +1,132 @@
+import type { ThreadMetadata } from '../../../liveblocks.config'
+import type { CanvasPoint, LocalPoint } from '../shared/math-utils'
+import { canvasPoint, localPoint } from '../shared/math-utils'
+import { assertNever } from '../shared/utils'
+
+export type UtopiaThreadMetadata = CanvasThreadMetadata | SceneThreadMetadata
+
+type BaseThreadMetadata = {
+  position: CanvasPoint
+  resolved: boolean
+}
+
+export type CanvasThreadMetadata = BaseThreadMetadata & {
+  type: 'canvas'
+}
+
+export type SceneThreadMetadata = BaseThreadMetadata & {
+  type: 'scene'
+  sceneId: string
+  scenePosition: LocalPoint
+  remixLocationRoute?: string
+}
+
+export function canvasThreadMetadata(
+  position: CanvasPoint,
+  resolved: boolean = false,
+): CanvasThreadMetadata {
+  return {
+    type: 'canvas',
+    position: position,
+    resolved: resolved,
+  }
+}
+
+export function sceneThreadMetadata(
```

Comment:
what about passing the parameters as an `Omit` for the `SceneThreadMetadata`, excluding the `type`? imho it would require less fiddling around going forward

```ts
function sceneThreadMetadata(data: Omit<SceneThreadMetadata, "type">) {
	return { ...data, type: 'scene' }
}
```

(same for the canvas metadata constructor)

Reply:
Fixed in 22f3b2b8b4afe10bd4e81515a7c806d3ffa08d61